bundles/org.openhab.binding.smaenergymeter/src/main/java/org/openhab/binding/smaenergymeter/internal/packet/PacketListener.java

```diff
@@ -0,0 +1,130 @@
+/**
+ * Copyright (c) 2010-2024 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.smaenergymeter.internal.packet;
+
+import java.io.IOException;
+import java.net.DatagramPacket;
+import java.net.DatagramSocket;
+import java.net.InetAddress;
+import java.net.MulticastSocket;
+import java.util.List;
+import java.util.concurrent.CopyOnWriteArrayList;
+import java.util.concurrent.ScheduledFuture;
+
+import org.openhab.binding.smaenergymeter.internal.handler.EnergyMeter;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The {@link PacketListener} class is responsible for communication with the SMA devices.
+ * It handles udp/multicast traffic and broadcast received data to subsequent payload handlers.
+ *
+ * @author Łukasz Dywicki - Initial contribution
+ */
+public class PacketListener {
+
+    private final DefaultPacketListenerRegistry registry;
+    private final List<PayloadHandler> handlers = new CopyOnWriteArrayList<>();
```

Comment:
It is better to use a `Set`, it is a precaution to avoid having the same object several times in the list.
You can find examples in other bindings. Let me know if you need help to find one.

Reply:
Please advice. i don't see any example.

Reply:
It was more an advice, not a request for change. It can work well with a List if the binding is properly implemented.
I remember we got memory leaks in certain bindings due to that. I have in mind I fixed that in elerotransmitterstick binding few years ago but I can't find the PR...
As it is just a suggestion, you can ignore it.